lib/extras.py

```diff
@@ -437,6 +438,150 @@ def callproc(self, procname, vars=None):
+
+        # prevent auto-issued BEGIN statements
+        if not self.async:
+            self.autocommit = True
+
+    def cursor(self, *args, **kwargs):
+        kwargs.setdefault('cursor_factory', ReplicationCursor)
+        return super(ReplicationConnection, self).cursor(*args, **kwargs)
+
+
+"""Streamging replication types."""
+REPLICATION_PHYSICAL = 0
+REPLICATION_LOGICAL  = 1
+
+class ReplicationCursor(_cursor):
+    """A cursor used for replication commands."""
+
+    def identify_system(self):
+        """Get information about the cluster status."""
+
+        self.execute("IDENTIFY_SYSTEM")
+        return dict(zip([_.name for _ in self.description],
+                        self.fetchall()[0]))
+
+    def quote_ident(self, ident):
+        # FIXME: use PQescapeIdentifier or psycopg_escape_identifier_easy, somehow
+        return '"%s"' % ident.replace('"', '""')
+
+    def create_replication_slot(self, slot_type, slot_name, output_plugin=None):
+        """Create streaming replication slot."""
+
+        command = "CREATE_REPLICATION_SLOT %s " % self.quote_ident(slot_name)
+
+        if slot_type == REPLICATION_LOGICAL:
+            if output_plugin is None:
+                raise psycopg2.ProgrammingError("output_plugin is required for logical replication slot")
+
+            command += "LOGICAL %s" % self.quote_ident(output_plugin)
+
+        elif slot_type == REPLICATION_PHYSICAL:
+            if output_plugin is not None:
+                raise psycopg2.ProgrammingError("output_plugin is not applicable to physical replication")
+
+            command += "PHYSICAL"
+
+        else:
+            raise psycopg2.ProgrammingError("unrecognized replication slot type")
+
+        self.execute(command)
+
+    def drop_replication_slot(self, slot_name):
+        """Drop streaming replication slot."""
+
+        command = "DROP_REPLICATION_SLOT %s" % self.quote_ident(slot_name)
+        self.execute(command)
+
+    def start_replication(self, slot_type, slot_name=None, writer=None, start_lsn=None,
+                          timeline=0, keepalive_interval=10, options=None):
+        """Start and consume replication stream."""
+
+        command = "START_REPLICATION "
+
+        if slot_type == REPLICATION_LOGICAL and slot_name is None:
+            raise psycopg2.ProgrammingError("slot_name is required for logical replication slot")
+
+        if slot_name:
+            command += "SLOT %s " % self.quote_ident(slot_name)
+
+        if slot_type == REPLICATION_LOGICAL:
+            command += "LOGICAL "
+        elif slot_type == REPLICATION_PHYSICAL:
+            command += "PHYSICAL "
+        else:
+            raise psycopg2.ProgrammingError("unrecognized replication slot type")
+
+        if start_lsn is None:
+            start_lsn = '0/0'
+
+        # reparse lsn to catch possible garbage
+        lsn = start_lsn.split('/')
+        command += "%X/%X" % (int(lsn[0], 16), int(lsn[1], 16))
```

Comment:
IMO the LSN should internally  be a uint64, formatted in Pg's x/y format for output on the wire.